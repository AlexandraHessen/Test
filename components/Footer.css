.Footer{
    display: flex;
    justify-content: center;
    width: 100%;
    color: #0a1006;
    font-size: 13px;
    backdrop-filter: blur(10px);
    background: #ffffff26;
    padding-bottom: 15px;
}

.Footer div{
    padding: 0 20px;
}

.Footer h3{
    padding: 10px 0;
    margin-bottom: 0;
    color: #8ccb68;
}

.Footer .MenuPages{
    font-size: 13px;
    

}

.Footer .PageLink{
    font-size: 13px;
    padding: 0;
    color: #0a1006;
}

.Footer .CategoryMenu{
    background: none;
}

.Footer .CategoryMenu li{
    padding: 0;
    border: none;
}

.Footer .CategoryMenu a {
    color: #0a1006;
}

.Footer .CategoryMenu{
    background: none;
}

.Footer  .ActivePageLink{
    background: none;
}

@media all and (max-width: 500px){
    .Footer h3{
        padding: 5px 0;
    }

    .Footer  .Menu{
        display: none;
    }
}