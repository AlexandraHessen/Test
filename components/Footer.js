import React from 'react';

import MenuPages from '../pages/MenuPages';
import CategoryMenu from './CategoryMenu';
import Contacts from './Contacts'

import './Footer.css'

class Footer extends React.PureComponent {
    render(){
        return(
            <footer className="Footer">
                <div className="Menu">
                    <h3>Меню</h3>
                    <MenuPages />
                </div>
                <div>
                    <h3>Контакты</h3>
                    <Contacts />
                </div>
            </footer>
        )
    }

}

export default Footer;