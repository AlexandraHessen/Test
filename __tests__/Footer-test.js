"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Footer from '../components/Footer';
                    //импортируем файл который будем тестировать

test('Footer', () => {
                            // создаём тестовую версию компонента который будем тестировать
  const component = renderer.create(
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  );

                            // получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
