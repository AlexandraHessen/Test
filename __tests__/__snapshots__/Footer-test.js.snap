// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
<footer
  className="Footer"
>
  <div
    className="Menu"
  >
    <h3>
      Меню
    </h3>
    <ul
      className="MenuPages"
    >
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/"
          onClick={[Function]}
          style={undefined}
        >
          Главная
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/catalog/1"
          onClick={[Function]}
          style={undefined}
        >
          Каталог
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/contacts"
          onClick={[Function]}
          style={undefined}
        >
          Контакты
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className="PageLink"
          href="/basket"
          onClick={[Function]}
          style={undefined}
        >
          Корзина
        </a>
      </li>
    </ul>
  </div>
  <div>
    <h3>
      Контакты
    </h3>
    <ul
      className="Contacts"
    >
      <li>
        +375 (29) 844-44-88
      </li>
      <li>
        +375 (44) 888-44-88
      </li>
      <li>
        Пн-Пт: 9:00-18:00
      </li>
      <li>
        г.Минск, пр.Победителей, 65
      </li>
    </ul>
  </div>
</footer>
`;
